sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/ArithmeticExpressionSuite.scala

```diff
@@ -505,4 +505,9 @@ class ArithmeticExpressionSuite extends SparkFunSuite with ExpressionEvalHelper
       checkEvaluation(e6, 0.toByte)
     }
   }
+
+  test("SPARK-31761: test integer overflow for (Divide) integral type ") {
+    checkEvaluation(IntegralDivide(Literal(Integer.MIN_VALUE), Literal(-1)), Integer
+      .MIN_VALUE.toLong * -1)
```

Comment:
Does it overflow in other DBMSes? If they do, I think we should just let it overflow and guide by ANSI configuration. see also https://github.com/apache/spark/commit/ee41001949af43d25dc6962ab6ca277e53c64299